File: core/src/main/java/net/dontdrinkandroot/wicket/behavior/StyleAppender.java
package net.dontdrinkandroot.wicket.behavior;

import net.dontdrinkandroot.wicket.model.StringModel;

import org.apache.wicket.behavior.AttributeAppender;
import org.apache.wicket.model.IModel;


/**
 * Appends the given style to the style attribute of an element.
 */
public class StyleAppender extends AttributeAppender {

	private static final long serialVersionUID = 793104976029930067L;


	public StyleAppender(final String style) {

		super("style", new StringModel(style), ";");
	}


	public StyleAppender(IModel<?> styleModel) {

		super("style", styleModel, ";");
	}

}

File: core/src/main/kotlin/net/dontdrinkandroot/wicket/behavior/StyleAppender.kt
package net.dontdrinkandroot.wicket.behavior

import org.apache.wicket.behavior.AttributeAppender
import org.apache.wicket.model.IModel
import org.apache.wicket.model.Model

/**
 * Appends the given style to the style attribute of an element.
 */
open class StyleAppender : AttributeAppender {

    constructor(style: String?) : super("style", Model(style), ";")

    constructor(styleModel: IModel<*>?) : super("style", styleModel, ";")
}